File: ament_package_example/src/topic_publisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

namespace ament_package_example {

  class TopicPublisher : public rclcpp::Node {
    public:
      TopicPublisher()
      : rclcpp::Node("topic_publisher")
      {
        sub_string_ = create_subscription<std_msgs::msg::String>("topic_in", 1, std::bind(&TopicPublisher::recv_string, this, std::placeholders::_1));
        pub_string_ = create_publisher<std_msgs::msg::String>("topic_out", 1);
      }

    private:
      rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub_string_;
      rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub_string_;

      void recv_string(const std_msgs::msg::String::ConstSharedPtr msg) {
        std_msgs::msg::String new_string;
        new_string.data = msg->data + "_123";
        pub_string_->publish(new_string);
      }
  };
}

int main(int argc, char** argv) {
  rclcpp::init(argc, argv);
  auto node_instance = std::make_shared<ament_package_example::TopicPublisher>();
  rclcpp::spin(node_instance);
  rclcpp::shutdown();
  return 0;
}

File: python_package_example/python_package_example/topic_publisher.py
#! /usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class TopicPublisher(Node):
    def __init__(self):
        super().__init__('topic_publisher')
        self.sub_string = self.create_subscription(topic='topic_in', msg_type=String, callback=self.recv_string, qos_profile=1)
        self.pub_string = self.create_publisher(topic='topic_out', msg_type=String, qos_profile=1)

    def recv_string(self, msg):
        new_string_msg = String()
        new_string_msg.data = msg.data + '_123'
        self.pub_string.publish(new_string_msg)


def main(args=None):
    rclpy.init()
    node_instance = TopicPublisher()
    rclpy.spin(node_instance)

if __name__ == '__main__':
    main()
